### PR TITLE
[fix](pipelinex) fix fragment instance progress reports (part 2)

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

This PR is a followup of #40325. Because PipelineX has deprecated the old report.
This PR fixed the `ScannedRows` and `LoadBytes` in the progress of the `SHOW LOAD` result.

Currently the progress will only be updated when a fragment instance is done.
Timely progress updates will be supported in a later PR.

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by liaoxin01** [comment]
Please add a regression test.

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 1]
```diff
@@ -2888,9 +2888,10 @@ public void unsetFields() {
         // Has to use synchronized to ensure there are not concurrent update threads. Or the done
         // state maybe update wrong and will lose data. see https://github.com/apache/doris/pull/29802/files.
         public synchronized boolean updatePipelineStatus(TReportExecStatusParams params) {
-            // The fragment or instance is not finished, not need update
+            // The fragment or instance is not finished, still need handle the report to update the load progress.
+            // Otherwise the load progress (ScannedRows/LoadBytes) won't be updated before any instance is done.
             if (!params.done) {
-                return false;
+                return true;
```
Comment (liaoxin01) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:2894]: This may bring some problems here, refer to the comments above.

Comment (kaijchen) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java]: reverted

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Pull request merged**